test/integration/kubernetes_deploy_test.rb

```diff
@@ -99,6 +99,54 @@ def test_pruning_disabled
     hello_cloud.assert_all_redis_resources_up
   end
 
+  def test_selector
+    ejson_cloud = FixtureSetAssertions::EjsonCloud.new(@namespace)
+    ejson_cloud.create_ejson_keys_secret
+
+    # Deploy the same thing twice with a different selector
+    assert_deploy_success(deploy_fixtures("branched",
+      bindings: { "branch" => "master" },
+      selector: { "branch" => "master" }))
+    assert_deploy_success(deploy_fixtures("branched",
+      bindings: { "branch" => "staging" },
+      selector: { "branch" => "staging" }))
+    deployments = v1beta1_kubeclient.get_deployments(namespace: @namespace, label_selector: "app=branched")
+
+    assert_equal 2, deployments.size
+    assert_equal %w(master staging), deployments.map { |d| d.metadata.labels.branch }.sort
+
+    # Run again without selector to verify pruning works
+    assert_deploy_success(deploy_fixtures("branched", bindings: { "branch" => "master" }))
+    deployments = v1beta1_kubeclient.get_deployments(namespace: @namespace, label_selector: "app=branched")
+
+    assert_equal 1, deployments.size
+    assert_equal "master", deployments.first.metadata.labels.branch
+  end
+
+  def test_mismatched_selector
+    assert_deploy_failure(deploy_fixtures("branched",
+      bindings: { "branch" => "master" },
+      selector: { "branch" => "staging" }))
+    assert_logs_match_all([
+      /Template validation failed/,
+      /Invalid template: Deployment/,
+      /selector branch=staging does not match labels app=branched,branch=master/,
```

Comment:
I see in the output that you pasted that this appears under "> Error from kubectl:", which is a bit confusing. Now that there are other sources of errors, let's make that more generic.

Reply:
There's another validator in kubernetes_resource.rb already, that has the same awkward result right now. I can think of a fix though

Reply:
True. Should be a matter of simply changing the string here though: https://github.com/Shopify/kubernetes-deploy/blob/master/lib/kubernetes-deploy/deploy_task.rb#L263

Reply:
I'll just change it to "> Error:", as it's already clear from the context that it's to do with template validation